File: app/src/main/java/com/example/saka/myapplication/adaper/OptionAdapter.java
package com.example.saka.myapplication.adaper;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.saka.myapplication.R;

/**
 * Created by Saka on 04-Jun-17.
 */

public class OptionAdapter extends BaseAdapter {

    private Context context;
    private String options[];
    private int icons[];

    public OptionAdapter(Context context, String[] options, int[] icons) {
        this.context = context;
        this.options = options;
        this.icons = icons;
    }

    public OptionAdapter(Context context, String[] options) {
        this.context = context;
        this.options = options;
    }

    @Override
    public int getCount() {
        return options.length;
    }

    @Override
    public Object getItem(int position) {
        return options[position];
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View rowView = LayoutInflater.from(context).inflate(R.layout.viewholder_option, parent, false);
            TextView txtOption = (TextView) rowView.findViewById(R.id.vh_option_txt);
            ImageView imgOption = (ImageView) rowView.findViewById(R.id.vh_option_img);
            txtOption.setText(options[position]);
            imgOption.setImageResource(icons[position]);
        convertView = rowView;
        return convertView;
    }

}

File: app/src/main/java/com/example/saka/myapplication/adaper/OptionAdapter.kt
package com.example.saka.myapplication.adaper

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.saka.myapplication.R

/**
 * Created by Saka on 04-Jun-17.
 */
class OptionAdapter(
    private var options: Array<String>,
    private var icons: IntArray
) : RecyclerView.Adapter<OptionAdapter.OptionHolder>() {

    inner class OptionHolder(view: View) : RecyclerView.ViewHolder(view) {


    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OptionHolder {
        val rowView =
            LayoutInflater.from(parent.context).inflate(R.layout.viewholder_option, parent, false)
        return OptionHolder(rowView)
    }

    override fun getItemCount(): Int {
        return options.size
    }

    override fun onBindViewHolder(holder: OptionHolder, position: Int) {
        holder.itemView.findViewById<TextView>(R.id.vh_option_txt).text = options[position]
        holder.itemView.findViewById<ImageView>(R.id.vh_option_img)
            .setImageResource(icons[position])
    }
}
